routers/api/v1/api.go

```diff
@@ -1195,6 +1195,7 @@ func Routes() *web.Router {
 					m.Get("/*", repo.GetBranch)
 					m.Delete("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.DeleteBranch)
 					m.Post("", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, bind(api.CreateBranchRepoOption{}), repo.CreateBranch)
+					m.Post("/{name}/rename", tokenRequiresScopes(auth_model.AccessTokenScopeCategoryRepository), reqRepoWriter(unit.TypeCode), bind(api.RenameBranchRepoOption{}), repo.RenameBranch)
```

Comment:
Maybe `/*/rename` can work? branch name maybe include `/`.

Reply:
I think I am missing something here, but wouldn't we expect `{name}` to be URL-encoded? This makes sense to me as it should avoid the http router from misinterpreting the path segments. The swagger client does URL-encode test/branch/rename when I make a request. Is this not something we expect clients to do?

Reply:
if `{name}` is URl-encoded, why not put them into the post key/value?

Reply:
If you are trying to follow GitHub, they do not need to encode that path.

And none of our handlers above requires to encode the "path".

![image](https://github.com/user-attachments/assets/d1fc779b-b23f-486e-b1ef-d039c0fe1704)

Reply:
Found this on GitHub's troubleshooting [page](https://docs.github.com/en/rest/using-the-rest-api/troubleshooting-the-rest-api?apiVersion=2022-11-28#404-not-found-for-an-existing-resource):
> Additionally, any path parameters must be URL encoded. For example, any slashes in the parameter value must be replaced with %2F. If you don't properly encode any slashes in the parameter name, the endpoint URL will be misinterpreted.

But I haven't seen this mentioned anywhere else when searching their docs, so not sure if it's true.

I also tested the [GitHub](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#rename-a-branch) rename branch API on a test repo using Postman (where I also disabled URL encoding my requests) and it looks like it still works?

I decided to go with the adding the old branch name as I realized that I previously used our swagger client to this the endpoint not realizing that `{name}` was URL-encoded; so when running the following `curl`:
```
curl -X 'POST' \
  'http://localhost:3000/api/v1/repos/korvax/pipdeptree/branches/test/branch/rename/rename' \
  -H 'accept: application/json' \
  -H 'authorization: Basic {ADD YOUR TOKEN HERE} \
  -H 'Content-Type: application/json' \
  -d '{
  "new_name": "dev"
}' --verbose
```

... would result in a 405 Method Not Allowed for me (I assume that go-chi is misinterpreting the path param?).